gluster/block/glusterblock-provisioner.go

```diff
@@ -161,25 +177,64 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 
 // createVolume creates a gluster block volume i.e. the storage asset.
 
-func (p *glusterBlockProvisioner) createVolume(PVName string) (string, string, error) {
-	var dtarget, diqn string
-	if p.provConfig.opMode == "executable" {
+func (p *glusterBlockProvisioner) createVolume(PVName string) (*glusterBlockVolume, error) {
+
+	switch p.provConfig.opMode {
+
+	case "executable":
 		cmd := exec.Command("sh", p.provConfig.execPath)
 		err := cmd.Run()
 		if err != nil {
 			glog.Errorf("%v", err)
 		}
-		dtarget = os.Getenv("TARGET")
-		diqn = os.Getenv("IQN")
+
+		dtarget := os.Getenv("TARGET")
+		diqn := os.Getenv("IQN")
+
+		p.volConfig.TargetPortal = dtarget
+		p.volConfig.IQN = diqn
+
+	case "rest":
+		cli := gcli.NewClient(p.provConfig.url, p.provConfig.user, p.provConfig.secretValue)
+		if cli == nil {
+			glog.Errorf("glusterfs: failed to create glusterfs rest client")
+			return nil, fmt.Errorf("glusterfs: failed to create glusterfs rest client, REST server authentication failed")
+		}
+
+		//TODO:
+		sz := 1
+		volumeReq := &gapi.VolumeCreateRequest{Size: sz}
+		_, err := cli.VolumeCreate(volumeReq)
+		if err != nil {
+			glog.Errorf("glusterfs: error creating volume %v ", err)
+			return nil, fmt.Errorf("error creating volume %v", err)
+		}
+
+		p.volConfig.IQN = "iqn.2016-12.org.gluster-block:aafea465-9167-4880-b37c-2c36db8562ea"
+		p.volConfig.TargetPortal = "192.168.1.11"
+
+	default:
+		return nil, fmt.Errorf("error parsing value for 'opmode' for volume plugin %s", provisionerName)
 	}
-	return dtarget, diqn, nil
+	return &p.volConfig, nil
+
+	/*
```

Comment:
delete this commented block? or does gofmt need to be run on this file, something about it looks off about it!